src/Components/Main/TodosContext/TodosContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo, useState } from 'react';
+import { ListAction } from '../../Enum/ListAction';
+import { Todos } from '../../Types';
+import { useLocalStorage } from '../../UseLocalStorage';
+
+type Props = {
+  children: React.ReactNode;
+};
+
+type TodosContext = {
+  todo: Todos[],
+  setTodo: (v: Todos[]) => void,
+  filter: ListAction,
+  setFilter: (v: ListAction) => void,
+  filterTodos: () => Todos[],
+  toggleAll: boolean;
+  setToggleAll: (ListAction: boolean) => void,
+};
+
+export const TodoContext = React.createContext<TodosContext>({
+  todo: [],
+  setTodo: () => {},
+  filter: ListAction.ALL,
+  setFilter: () => {},
+  filterTodos: () => [],
+  toggleAll: false,
```

Comment:
toggleAll sounds as a function, it's better to rename it to allToggled or isAllToggled

Reply:
not fixed